npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -69,6 +69,10 @@ function modifyWebpackConfigForCypress (webpackConfig: Partial<Configuration>) {
     }
   }
 
+  if (webpackConfig?.module?.rules) {
```

Comment:
Maybe we should leave a comment here about why we're doing this? I also want to make sure we're all in agreement that this is the right way to go considering that it could have a potentially large effect on bundle sizes. @ZachJW34 what are your thoughts?

Reply:
I have added a comment on my latest commit. Also:
Based on conversations I've had with @astone123 and @ZachJW34  there seems to be two possible ways to tackling this as summarized below.
- The first is to introduce a new rule in the user's `webpack config` targeting all major styling files e.g. `.css, .scss, .sass, .styl, .less` and setting the `sideEffect: true` for them. The benefit of this solution is that we guarantee that sideEffects are set for the above files as we specifically target them in the new rule we add .
- The second direction would be to modify the existing rules in `webpack config` by setting `sideEffects: true` for any existing rules that are loading styling files. The benefit is that this could be considered 'safer' as we do not add any new rules. However, there is a higher chance of missing styling files especially when the `rules.test` set by the user are complicated and don't directly refer to the file extensions.

I have gone with the first option but I'm definitely interested in everyone's thoughts